### PR TITLE
Misc compatibility updates

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): https://github.com/unoplatform/private/issues/6

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the new behavior?

- Enumerate CompositionTarget handlers safely
- Add non-failing StatusBar BackgroundOpacity and BackgroundColor getters.
- Relax DependencyProperty owner validation for non-FrameworkElement
- [Wasm] Fix TextBoxView SelectionStart/SelectionEnd value parsing
- [Wasm] Don't fail on FrameworkElement.Dispose()

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.wasm.cs [hunk 1]
```diff
@@ -88,13 +89,35 @@ internal void SetEnabled(bool newValue)
 
 		public int SelectionStart
 		{
-			get => int.Parse(GetProperty("selectionStart"));
+			get
+			{
+				if (int.TryParse(GetProperty("selectionStart"), NumberStyles.Integer, CultureInfo.InvariantCulture, out var result))
```
[the remark this answers is not in the document]
Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.wasm.cs]:
```suggestion
				get => int.TryParse(GetProperty("selectionStart"), NumberStyles.Integer, CultureInfo.InvariantCulture, out var result) ? result : 0;
```
😊

[the remark this answers is not in the document]
Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.wasm.cs]: BTW "Some cases", do you have any idea when ?

File: src/Uno.UI/UI/Xaml/DependencyObjectStore.cs [hunk 2]
```diff
@@ -583,7 +583,17 @@ private void ValidatePropertyOwner(DependencyProperty property)
 		{
 			if (_validatePropertyOwner)
 			{
-				if (!_originalObjectType.Is(property.OwnerType) && !property.IsAttached)
+				var iFrameworkElement = _originalObjectType.Is(typeof(FrameworkElement));
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/DependencyObjectStore.cs]:
```suggestion
				var isFrameworkElement = _originalObjectType.Is(typeof(FrameworkElement));
```

File: src/Uno.UI/UI/Xaml/DependencyObjectStore.cs [hunk 3]
```diff
@@ -583,7 +583,17 @@ private void ValidatePropertyOwner(DependencyProperty property)
...
+				var isMixinFrameworkElement = _originalObjectRef.Target is IFrameworkElement && !iFrameworkElement;
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/DependencyObjectStore.cs]:
```suggestion
				var isMixinFrameworkElement = _originalObjectRef.Target is IFrameworkElement && !isFrameworkElement;
```

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.wasm.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.wasm.cs]: Not at this point, will know soon.

[the remark this answers is not in the document]
Comment (carldebilly) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.wasm.cs]: empty string?

File: src/Uno.UI/UI/Xaml/Media/CompositionTarget.iOS.cs [hunk 4]
```diff
@@ -40,7 +41,7 @@ public  partial class CompositionTarget
 
 		private static void OnFrame()
 		{
-			_handlers.ForEach(h => h(null, null));
+			_handlers.ToList().ForEach(h => h(null, null));
```
Comment (carldebilly) [inline src/Uno.UI/UI/Xaml/Media/CompositionTarget.iOS.cs]: Why do you force a materialization of the list here?

Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Media/CompositionTarget.iOS.cs]: There's a race condition with the handlers list. It's not used much there yet.

**Pull request merged**